pkg/gohai/cpu/cpu_windows.go

```diff
@@ -59,96 +46,110 @@ type SYSTEM_INFO struct {
+	matches := re.FindStringSubmatch(caption)
+	if len(matches) > 0 {
+		return strings.Split(matches[0], " ")[1]
 	}
-	return
-}
-
-func getSystemInfo() (si SYSTEM_INFO) {
-	var mod = windows.NewLazyDLL("kernel32.dll")
-	var gsi = mod.NewProc("GetSystemInfo")
-
-	// syscall does not fail
-	//nolint:errcheck
-	gsi.Call(uintptr(unsafe.Pointer(&si)))
-	return
+	return ""
 }
 
 func getCPUInfo() *Info {
-	cpuInfo := &Info{
-		CacheSizeKB: utils.NewErrorValue[uint64](utils.ErrNotCollectable),
+	var cInfo cpuInfo
+	var cCPUInfo C.CPU_INFO
+
+	ret := C.computeCoresAndProcessors(&cCPUInfo)
+	if ret != 0 {
+		log.Errorf("failed to get CPU information, error code: %d", ret)
+		return &Info{}
 	}
 
+	// Copy C struct values to Go struct
+	cInfo.corecount = int(cCPUInfo.corecount)
+	cInfo.logicalcount = int(cCPUInfo.logicalcount)
+	cInfo.pkgcount = int(cCPUInfo.pkgcount)
+	cInfo.numaNodeCount = int(cCPUInfo.numaNodeCount)
+	cInfo.relationGroups = int(cCPUInfo.relationGroups)
+	cInfo.maxProcsInGroups = int(cCPUInfo.maxProcsInGroups)
+	cInfo.activeProcsInGroups = int(cCPUInfo.activeProcsInGroups)
+	cInfo.l1CacheSize = uint64(cCPUInfo.l1CacheSize)
+	cInfo.l2CacheSize = uint64(cCPUInfo.l2CacheSize)
+	cInfo.l3CacheSize = uint64(cCPUInfo.l3CacheSize)
+
+	// Get additional CPU information from Windows registry
 	k, err := registry.OpenKey(registry.LOCAL_MACHINE,
 		registryHive,
 		registry.QUERY_VALUE)
-	if err == nil {
-		defer k.Close()
-
-		dw, _, err := k.GetIntegerValue("~MHz")
-		cpuInfo.Mhz = utils.NewValueFrom(float64(dw), err)
-
-		s, _, err := k.GetStringValue("ProcessorNameString")
-		cpuInfo.ModelName = utils.NewValueFrom(s, err)
-
-		s, _, err = k.GetStringValue("VendorIdentifier")
-		cpuInfo.VendorID = utils.NewValueFrom(s, err)
-
-		s, _, err = k.GetStringValue("Identifier")
-		if err == nil {
-			cpuInfo.Family = utils.NewValue(extract(s, "Family"))
-		} else {
-			cpuInfo.Family = utils.NewErrorValue[string](err)
-		}
-	} else {
-		cpuInfo.Mhz = utils.NewErrorValue[float64](err)
-		cpuInfo.ModelName = utils.NewErrorValue[string](err)
-		cpuInfo.VendorID = utils.NewErrorValue[string](err)
-		cpuInfo.Family = utils.NewErrorValue[string](err)
+	if err != nil {
+		log.Errorf("failed to open registry key: %v", err)
+		return &Info{}
 	}
+	defer k.Close()
 
-	cpus, cpuerr := computeCoresAndProcessors()
-	cpuInfo.CPUPkgs = utils.NewValueFrom(uint64(cpus.pkgcount), cpuerr)
-	cpuInfo.CPUNumaNodes = utils.NewValueFrom(uint64(cpus.numaNodeCount), cpuerr)
-	cpuInfo.CPUCores = utils.NewValueFrom(uint64(cpus.corecount), cpuerr)
-	cpuInfo.CPULogicalProcessors = utils.NewValueFrom(uint64(cpus.logicalcount), cpuerr)
-	cpuInfo.CacheSizeL1Bytes = utils.NewValueFrom(uint64(cpus.l1CacheSize), cpuerr)
-	cpuInfo.CacheSizeL2Bytes = utils.NewValueFrom(uint64(cpus.l2CacheSize), cpuerr)
-	cpuInfo.CacheSizeL3Bytes = utils.NewValueFrom(uint64(cpus.l3CacheSize), cpuerr)
+	if mhz, _, err := k.GetIntegerValue("~MHz"); err == nil {
+		cInfo.Mhz = float64(mhz)
+	}
```

Comment:
How about log.Error if the err is not nil? Same for the next few lines. If there are known cases where this fails, and we don't want to be too noisy, maybe explicitly check for them.